go/pkg/depgraph/builder.go

```diff
@@ -99,10 +101,14 @@ func (b *Builder) GetRootNode() *Node {
 }
 
 func (b *Builder) AddNode(nodeID string, pkgInfo *PkgInfo) *Node {
-	return b.addNode(nodeID, pkgInfo)
+	return b.addNode(nodeID, pkgInfo, nil)
 }
 
-func (b *Builder) addNode(nodeID string, pkgInfo *PkgInfo) *Node {
+func (b *Builder) AddNodeWithInfo(nodeID string, pkgInfo *PkgInfo, nodeInfo *NodeInfo) *Node {
+	return b.addNode(nodeID, pkgInfo, nodeInfo)
+}
```

Comment:
Adding to what @gitphill said, I would love it if we could make `AddNode` variadic instead.

```go
// Something like this
type opt = func(*Node)

func WithNodeInfo(info *NodeInfo) opt {
	return func(n *Node) {
		n.Info = info
	}
}

func (b *Builder) AddNode(nodeID string, pkgInfo *PkgInfo, ...opt) *Node {
	...
}
```